pkg/server/sandbox_status.go

```diff
@@ -66,26 +66,38 @@ func (c *criService) PodSandboxStatus(ctx context.Context, r *runtime.PodSandbox
 	}, nil
 }
 
-func (c *criService) getIP(sandbox sandboxstore.Sandbox) (string, error) {
+func (c *criService) getNetworkStatus(sandbox sandboxstore.Sandbox) (*runtime.PodSandboxNetworkStatus, error) {
 	config := sandbox.Config
+	ns := runtime.PodSandboxNetworkStatus{}
 
 	if config.GetLinux().GetSecurityContext().GetNamespaceOptions().GetNetwork() == runtime.NamespaceMode_NODE {
 		// For sandboxes using the node network we are not
 		// responsible for reporting the IP.
-		return "", nil
+		return &ns, nil
 	}
 
 	if closed, err := sandbox.NetNS.Closed(); err != nil {
-		return "", errors.Wrap(err, "check network namespace closed")
+		return nil, errors.Wrap(err, "check network namespace closed")
 	} else if closed {
-		return "", nil
+		return &ns, nil
 	}
 
-	return sandbox.IP, nil
+	ns.Ip = sandbox.IP
+	// add additional ips assigned from CNI
+	if configs, ok := sandbox.CNIResult.Interfaces[defaultIfName]; ok && len(configs.IPConfigs) > 0 {
```

Comment:
are we guaranteed that `config != nil`?

Reply:
Yes, when the pod is being created, in [`setupPod`](https://github.com/kevpar/cri/blob/cb50770a95502a1d6714e0a1b61b8f3893b41ea2/pkg/server/sandbox_run.go#L82-L89), if `len(configs) == 0`, the pod will fail to create since it needs at least 1 IP address. Since `config` is `[]*cni.IPConfig`, being `nil` will fail that check and we will never get to checking for the status.